js/containers/bottom-controls.js

```diff
@@ -165,6 +222,29 @@ class BottomControls extends PureComponent {
     log('ResetClicked')
   }
 
+  handleRecentToggle (e) {
+    const { filters, setFilter, setEarthquakesVisible, setEruptionsVisible, mapRegion, mapZoom } = this.props
+    const currentDate = Date.now()
+    const monthAgo = currentDate - thirtyDays
+    // toggle between original start/end and the "recent 30 day" view
+    const startDate = filters.get('minTimeLimit') !== filters.get('initialStartTime') ? filters.get('initialStartTime') : monthAgo
+    const endDate = filters.get('maxTimeLimit') !== filters.get('initialEndTime') ? filters.get('initialEndTime') : currentDate
+
+    setFilter('minTime', startDate)
+    setFilter('maxTime', config.earthquakesDisplayAllOnStart ? endDate : startDate)
+
+    setFilter('minTimeLimit', startDate)
+    setFilter('maxTimeLimit', endDate)
+
+    setFilter('playbackMaxTime', endDate)
+
+    // if the box is checked, set both earthquakes and eruptions visible
```

Comment:
Was that a new requirement? I could imagine the project team wanted that, although it feels a bit weird that this checkbox has this side-effect. I bet we have some URL configs etc. where it can make things a bit weird.

Maybe we just should show eruptions by default instead?

Reply:
These came in response from the team - if the software isn't set to show all on start, we shouldn't change the behavior when checking the box. And if we're using the "last 30 days" view then it's more for the immediate lookup of information by casual users than from classroom setting where the default is to show the time slider over a longer period.